.github/workflows/CI_build.yml

```diff
@@ -7,6 +7,36 @@ on:
 
 jobs:
 
+  thorough_check:
```

Comment:
I did not look into the workflows, since I do not have enough experience with this to say anything useful about these changes.

Reply:
That's fine. It should have been separate anyway, but I mostly copied it from another repo where it had been reviewed.